middle_end/flambda2/types/expand_head.ml

```diff
@@ -491,24 +482,18 @@ let make_suitable_for_environment env (to_erase : to_erase) bind_to_and_types =
           (fun (bind_to, ty) ->
             let bind_to' = Renaming.apply_name renaming bind_to in
             let ty = TG.apply_renaming ty renaming in
-            let fresh_var =
-              Name.pattern_match bind_to'
-                ~var:(fun var ->
-                  if not (Name.equal bind_to bind_to') then Some var else None)
-                ~symbol:(fun _ -> None)
-            in
-            bind_to', fresh_var, ty)
+            bind_to', ty)
           equations
       in
       (* Finally form an environment extension with extra variables: the
          existentials are the fresh variables. *)
       List.fold_left
-        (fun env_extension (bind_to, fresh_var, ty) ->
+        (fun env_extension (bind_to, ty) ->
           let env_extension =
-            match fresh_var with
-            | Some bind_to ->
-              TEEV.add_definition env_extension bind_to (TG.kind ty)
-            | None -> env_extension
+            Name.pattern_match bind_to
+              ~var:(fun bind_to ->
+                TEEV.add_definition env_extension bind_to (TG.kind ty))
```

Comment:
Don't you need a distinction between the original "bind-to" variables (which should be already defined in the target environment) and the unavailable variables ?

Reply:
Ah yes, whoops.  Should be fixed now.